src/backend/base/langflow/api/build.py

```diff
@@ -0,0 +1,443 @@
+            inactivated_vertices = list(graph.inactivated_vertices)
+            graph.reset_inactivated_vertices()
+            graph.reset_activated_vertices()
+            # graph.stop_vertex tells us if the user asked
+            # to stop the build of the graph at a certain vertex
+            # if it is in next_vertices_ids, we need to remove other
+            # vertices from next_vertices_ids
+            if graph.stop_vertex and graph.stop_vertex in next_runnable_vertices:
+                next_runnable_vertices = [graph.stop_vertex]
+
+            if not graph.run_manager.vertices_being_run and not next_runnable_vertices:
+                background_tasks.add_task(graph.end_all_traces)
+
+            build_response = VertexBuildResponse(
+                inactivated_vertices=list(set(inactivated_vertices)),
+                next_vertices_ids=list(set(next_runnable_vertices)),
+                top_level_vertices=list(set(top_level_vertices)),
+                valid=valid,
+                params=params,
+                id=vertex.id,
+                data=result_data_response,
+            )
+            background_tasks.add_task(
+                telemetry_service.log_package_component,
+                ComponentPayload(
+                    component_name=vertex_id.split("-")[0],
+                    component_seconds=int(time.perf_counter() - start_time),
+                    component_success=valid,
+                    component_error_message=error_message,
+                ),
+            )
+        except Exception as exc:
+            background_tasks.add_task(
+                telemetry_service.log_package_component,
+                ComponentPayload(
+                    component_name=vertex_id.split("-")[0],
+                    component_seconds=int(time.perf_counter() - start_time),
+                    component_success=False,
+                    component_error_message=str(exc),
+                ),
+            )
+            logger.exception("Error building Component")
+            message = parse_exception(exc)
+            raise HTTPException(status_code=500, detail=message) from exc
+
+        return build_response
+
+    async def build_vertices(
+        vertex_id: str,
+        graph: Graph,
+        event_manager: EventManager,
+    ) -> None:
+        """Build vertices and handle their events.
+
+        Args:
+            vertex_id: The ID of the vertex to build
+            graph: The graph instance
+            client_consumed_queue: Queue for tracking client consumption
+            event_manager: Manager for handling events
+        """
+        build_task = asyncio.create_task(_build_vertex(vertex_id, graph, event_manager))
+        try:
+            await build_task
+            vertex_build_response: VertexBuildResponse = build_task.result()
+        except asyncio.CancelledError:
+            logger.debug("Build task cancelled")
+            build_task.cancel()
+            return
```

Comment:
I think this could just be
```suggestion
        vertex_build_response = await _build_vertex(vertex_id, graph, event_manager))
```
AFAIU, if `build_vertices` is cancelled, it will propagate to `_build_vertex`